src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -134,16 +140,25 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
 
         allocate_tensor(outputName, resultDescriptor, TensorType::InputOrOutput, allocator);
 
-        if (contains(_metadata.shapeNames, outputName)) {
-            const std::string shapeBufferName = SHAPE_TENSOR_PREFIX + outputName;
-            const IONodeDescriptor& shapeDescriptor = _metadata.shapes.at(outputName);
+        const auto& shapeNameMatch = _nodeNameToLegacyName.find(outputName);
+        if (shapeNameMatch != _nodeNameToLegacyName.end()) {
+            if (contains(_metadata.shapeNames, shapeNameMatch->second)) {
+                const std::string shapeBufferName = SHAPE_TENSOR_PREFIX + shapeNameMatch->second;
+                const IONodeDescriptor& shapeDescriptor = _metadata.shapes.at(shapeNameMatch->second);
 
-            check_level_zero_attributes_match(shapeDescriptor,
-                                              executorOutputDescriptors.at(shapeBufferName),
-                                              shapeBufferName);
+                check_level_zero_attributes_match(shapeDescriptor,
+                                                  executorOutputDescriptors.at(shapeBufferName),
+                                                  shapeBufferName);
 
-            auto allocator = zeroMemory::HostMemAllocator(backendPtr);
-            allocate_tensor(outputName, shapeDescriptor, TensorType::Shape, allocator);
+                ov::Allocator allocator;
```

Comment:
Can you consider reusing a single allocator for the whole constructor?

Reply:
Using only one for states and outputs but still need another one for inputs. It needs to be created with different flags for inputs.